src/rgw/rgw_string.cc

```diff
@@ -3,6 +3,19 @@
 
 #include "rgw_string.h"
 
+namespace detail {
+
+void gen_rand_buffer_mutate_in_place(CryptoRandom& cr, char *out, const size_t nchars, std::string_view tbl)
+{
+ cr.get_bytes(out, nchars);
+
+ for (size_t i = 0; nchars != i; ++i) {
+     *(i + out) = tbl[tbl[*(i + out)] & tbl.size()];
```

Comment:
i'm having trouble parsing this, can you maybe add a comment explaining what's going on here? how can you guarantee that the inner `tbl[*(i + out)]` doesn't overflow, and why is it using a character from `tbl` as another index into `tbl`?

Reply:
You know, I can only figure that I adapted it that way for some reason but I've looked and looked and can't fathom it myself-- I've reverted it to following along with the previous implementation, which also looks more correct to me! Thanks for commenting on this headscratcher!